it/__init__.py

```diff
@@ -0,0 +1,187 @@
+# Licensed to Elasticsearch B.V. under one or more contributor
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#	http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import functools
+import json
+import os
+import random
+
+import pytest
+
+from esrally import client
+from esrally.utils import process, io
+
+CONFIG_NAMES = ["in-memory-it", "es-it"]
+
+
+def all_rally_configs(t):
+    @functools.wraps(t)
+    @pytest.mark.parametrize("cfg", CONFIG_NAMES)
+    def wrapper(cfg, *args, **kwargs):
+        t(cfg, *args, **kwargs)
+
+    return wrapper
+
+
+def random_rally_config(t):
+    @functools.wraps(t)
+    @pytest.mark.parametrize("cfg", [random.choice(CONFIG_NAMES)])
+    def wrapper(cfg, *args, **kwargs):
+        t(cfg, *args, **kwargs)
+
+    return wrapper
+
+
+def rally_in_mem(t):
+    @functools.wraps(t)
+    @pytest.mark.parametrize("cfg", ["in-memory-it"])
+    def wrapper(cfg, *args, **kwargs):
+        t(cfg, *args, **kwargs)
+
+    return wrapper
+
+
+def rally_es(t):
+    @functools.wraps(t)
+    @pytest.mark.parametrize("cfg", ["es-it"])
+    def wrapper(cfg, *args, **kwargs):
+        t(cfg, *args, **kwargs)
+
+    return wrapper
+
+
+def esrally(cfg, command_line):
+    return os.system("esrally {} --configuration-name=\"{}\"".format(command_line, cfg))
+
+
+def wait_until_port_is_free(port_number=39200):
+    import errno
+    import time
+    import socket
+    while True:
+        c = socket.socket()
+        connect_result = c.connect_ex(("127.0.0.1", port_number))
+        # noinspection PyBroadException
+        try:
+            if connect_result == errno.ECONNREFUSED:
+                c.close()
+                return
+            else:
+                c.close()
+                time.sleep(0.5)
```

Comment:
Not directly related to this PR, but going forwards should we start embracing the https://github.com/jd/tenacity library, which seems to be becoming the standard in retrying in Python? It offers a lot flexibility in what to retry with boolean operations.

Reply:
I am reluctant adding additional dependencies for such small pieces of functionality but I added a two minute timeout in https://github.com/elastic/rally/pull/919/commits/3537d0bb657f94ea211ab4a05e3029cb8923b3f4.